src/z3c/recipe/sphinxdoc/__init__.py

```
@@ -168,9 +171,11 @@ def install(self):
     update = install
 
 
-def main(projects):
+def main(projects, argv=None, exit_on_error=False):
```

Shouldn't exit_on_error default to True?

---

I don't *think* so? This recipe is used for generating the docs for many projects at once. I don't think you want it to bail after generating doc number 3 of 50 just because doc number 3 happened to have an error.

It didn't behave that way in the past, AFAICS.

---

I don't have an issue for this yet, but our fork of this project goes a step further and needs to handle [bad eggs at script creation time](https://github.com/NextThought/z3c.recipe.sphinxdoc/commit/a09782462f00993817d4f4dc1e35826b275d9177). 

I think this recipe is a "best effort" recipe, not a "fail fast" or "bondage-and-discipline-warnings-are-errors" recipe.

---

I didn't check, but I thought `sphinx.main()` that was there before would exit with an non-zero exit code on error?

As far as I can see, this `main()` is invoked as an entry point, from a new script generated in `bin/`, and I think this script should exit with non-zero if documentation generation fails.

---

I was wrong: `sphinx.main()` doesn't call `sys.exit()`, it just returns the result of `sphinx.build_main()`.

Never mind!

---

There was actually a brief period of time where `sphinx.main` *did* exit. That introduced a bug in this package (#2). But later sphinx removed it again.

---

> and I think this script should exit with non-zero if documentation generation fails.

I wouldn't argue with that, *if* it tried to generate all the docs before exiting. So something like:
```python
code = 0
for project in projects:
   code |= sphinx.build()
sys.exit(code)
```

(Except I find it obnoxious when functions call `sys.exit` because it makes them annoying to unittest. I tend to reserve that for the `if __name__ == '__main__':` block. Which doesn't quite work here.)